### ISSUE TITLE
New lines are removed from docstrings if prefixed by "@brief" Doxygen style

### ISSUE THREAD

**Issue by Delgan** [microsoft/vscode-cpptools#10039]
### Environment

- OS and version: Ubutnu 20.04
- VS Code: 1.72.2
- C/C++ extension: v1.12.4
- OS and version of remote machine (if applicable): Ubuntu 20.04 (dev container)
- GDB / LLDB version: N/A

### Bug Summary and Steps to Reproduce

**Bug Summary:**

When a docstrings starts with `@brief` (which is a Doxygen convention), then newlines are ignored. This is not very convenient, because the `@brief` should appear on the first line and detailed description on following blocks.

```cpp
/**
 * @brief A brief description.
 * 
 * Detailed description starts here.
 */
void foobar() {}
```

With `@brief`:

![with_brief](https://user-images.githubusercontent.com/4193924/196959897-377f6580-a04e-4991-baab-4b32a927f2a0.png)

Without `@brief` (expected):

![without_brief](https://user-images.githubusercontent.com/4193924/196960013-0eb5775e-250e-44f3-a365-9f5f46a38614.png)

### Debugger Configurations

```shell
N/A
```

### Debugger Logs

```shell
N/A
```

**Comment by Delgan** [comment]
It seems it's a regression, it was already fixed with:
* https://github.com/microsoft/vscode-cpptools/issues/5741

**Comment by sean-mcmanus** [comment]
It's not a regression -- the fix was to turn the newline into a space.

**Comment by sean-mcmanus** [comment]
This was opened on VS at https://developercommunity.visualstudio.com/t/Tooltips-are-not-showing-doxygen-comment/10087897 . They closed it as "not a bug". You could try filing a feature request.

**Comment by Delgan** [comment]
@sean-mcmanus Indeed, sorry for calling this a regression. I thought it worked some times ago, I must have been dreaming.

Following your advice, I opened a "feature request" ticket here: [Avoid converting newlines into spaces in Doxygen "brief" comments (#10178887)](https://developercommunity.visualstudio.com/t/Avoid-converting-newlines-into-spaces-in/10178887). :+1:

**Comment by dimsanius** [comment]
I have realised its not just the `@brief` that's causing the problem... Instead, it is any type of Doxygen specific keyword (i.e. `@param`, `@note`, `@return`, `@returns`)

**Comment by starball5** [comment]
Related on Stack Overflow: [Line breaks are not preserved following @brief and similar Doxygen keywords in the VS Code hover info with the Cpptools extension](https://stackoverflow.com/q/76471935/11107541).

**Comment by JPHutchins** [comment]
> @sean-mcmanus Indeed, sorry for calling this a regression. I thought it worked some times ago, I must have been dreaming.
> 
> Following your advice, I opened a "feature request" ticket here: [Avoid converting newlines into spaces in Doxygen "brief" comments (#10178887)](https://developercommunity.visualstudio.com/t/Avoid-converting-newlines-into-spaces-in/10178887). 👍

Make sure to upvote here: https://developercommunity.visualstudio.com/t/Avoid-converting-newlines-into-spaces-in/10178887

IMO it's a bug.

**Comment by bastian2001** [comment]
An annoying workaround (setup process and only one new line) until this is pushed to release is to use the @detail command to get a line break between the brief and description. Make sure to enable the details tag in Extensions -> C/C++ -> Code Documentation -> Section Tags (`C_Cpp.doxygen.sectionTags`)

**Comment by BA7LYA** [comment]
This problem still exist.

**Comment by emrekz** [comment]
A another workaround which may helps but has not regular font style.
![output](https://github.com/user-attachments/assets/4e63cee9-ba48-4a89-a28d-b1a15c418057)

```c
/**
 * @brief #### Test description 1
 * @brief #### Test description 2
 * @brief #### Test description 3
 * @param x integer
 * @param y integer
 */
void testFunction() {
...
}
```

**Comment by bobbrow** [comment]
I believe this is fixed starting in 1.21.1 pre-release (soon to be published). We will interpret an empty line as a paragraph break.

```cpp
/**
 * @brief Hello there 
 * 
 * Hello again
 */
void foo();
```
![image](https://github.com/user-attachments/assets/c41c6498-8ba1-41c0-a2b2-e3f6b256adc2)

**Comment by JPHutchins** [comment]
> I believe this is fixed starting in 1.21.1 pre-release (soon to be published). We will interpret an empty line as a paragraph break.
> 
> ```c++
> /**
>  * @brief Hello there 
>  * 
>  * Hello again
>  */
> void foo();
> ```
> 
> ![image](https://private-user-images.githubusercontent.com/12818240/348778208-c41c6498-8ba1-41c0-a2b2-e3f6b256adc2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI2NDI1MDAsIm5iZiI6MTcyMjY0MjIwMCwicGF0aCI6Ii8xMjgxODI0MC8zNDg3NzgyMDgtYzQxYzY0OTgtOGJhMS00MWMwLWEyYjItZTNmNmIyNTZhZGMyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODAyVDIzNDMyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJiZGVmNDJiMDQ1YzYxNGEzYjRhNWRlOTQzMWY0ZWY2MjgwYzY3M2Q5Mjc2OGQ4NzhmYmUwZWYyOTRmMDNkNzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-Hl2jRY72IYat-O4krs-GSscQm5_PYfza86ohBOF0Eo)

It's fixed!  I'm crying.  🥲

```c
/**
 * @brief Encode the response into a packet.
 * 
 * Newlines work!  If there's a blank line first!
 * @param response The response to encode.
 * @param [out] packet Pointer to the output packet.
 * @return Error code.
 */
int api_encode(struct api_response const * response, union api_packet * out_packet);
```

<img width="355" alt="image" src="https://github.com/user-attachments/assets/4d601e36-a650-48a0-a42b-463617884406">

**Comment by bobbrow** [comment]
@JPHutchins from your screenshot it looks like you could benefit from #5773 as well.  Did you add your 👍 to it yet?

**Comment by Delgan** [comment]
> I believe this is fixed starting in 1.21.1 pre-release (soon to be published). We will interpret an empty line as a paragraph break.
> 
> ```c++
> /**
>  * @brief Hello there 
>  * 
>  * Hello again
>  */
> void foo();
> ```
> 
> ![image](https://private-user-images.githubusercontent.com/12818240/348778208-c41c6498-8ba1-41c0-a2b2-e3f6b256adc2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc3ODQ1MDcsIm5iZiI6MTcyNzc4NDIwNywicGF0aCI6Ii8xMjgxODI0MC8zNDg3NzgyMDgtYzQxYzY0OTgtOGJhMS00MWMwLWEyYjItZTNmNmIyNTZhZGMyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDAxVDEyMDMyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZiZmFhMjM4MjM4OWRiMTM0ZTRkYzgwYzkxYmVjYzY2OWE1ZTljNzY3ZjRlZTMwNDc3YjU0ZDQ1ZjI5ZGI0M2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.nghcSX9O4xjkuDn3Oct0eow6yHp3ZIxaNaIBOJHKl4E)

@bobbrow This is great, I can confirm that it works very well, thanks!

I made some tests and discovered, however, that it breaks the rendering of code blocks (v1.21.6):

```cpp
/**
 * @brief Hello there
 *
 * Hello again.
 *
 * ```cpp
 * int main() {
 *     std::cout << "Hello from code\n";
 *     return 1;
 * }
 *```
 */
void foo();
```
![Screenshot from 2024-10-01 14-07-36](https://github.com/user-attachments/assets/7301dd0c-d535-409b-ad39-cf2ea4f28646)

Should I open a new issue?

**Comment by bobbrow** [comment]
@Delgan, I can open it for you. Thanks for reporting it. This issue was supposed to have been closed with the release of 1.21.
